File: src/modules/planet/planet-model.js
const mongoose = require("mongoose");

const planetSchema = new mongoose.Schema({
  name: {
    type: String,
    required: true,
    trim: true,
    unique: true,
  },
  rotation_period: {
    type: String,
    default: "12",
  },
  orbital_period: {
    type: String,
    default: "365",
  },
  diameter: {
    type: String,
    default: "12742",
  },
  climate: {
    type: String,
  },
  gravity: {
    type: String,
  },
  terrain: {
    type: String,
  },
  surface_water: {
    type: String,
    default: "8",
  },
  population: {
    type: String,
    default: "70000000000",
  },
  residents: {
    type: [String],
  },
  films: {
    type: [String],
  },
  is_deleted: {
    type: Boolean,
  },
});

const Planet = mongoose.model("Planet", planetSchema);

module.exports = Planet;

File: csharp-asp/modules/planet/planet-model.cs
using MongoDB.Bson;
using System.Reflection;
using System.Security.Cryptography;

namespace csharp_asp.modules.planet
{
    public class Planet: IPlanet
    {
        public string _id { get; set; }
        public string Name { get; set; }
        public string? RotationPeriod { get; set; }
        public string? OrbitalPeriod { get; set; }
        public string? Diameter { get; set; }
        public string? Climate { get; set; }
        public string? Gravity { get; set; }
        public string? Terrain { get; set; }
        public string? SurfaceWater { get; set; }
        public string? Population { get; set; }
        public List<string>? Residents { get; set; }
        public List<string>? Films { get; set; }
        public Boolean IsDeleted { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set;}

        public Planet()
        {
            _id = ObjectId.GenerateNewId().ToString();
            Name = "NewEarth";
            RotationPeriod = "365";
            OrbitalPeriod = "365";
            Diameter = "500000";
            Climate = "Tempered";
            Gravity = "10";
            Terrain = "Plain";
            SurfaceWater = "Yes";
            Population = "10000000";
            IsDeleted = false;
        }
    }
}
